client/src/components/ComboBox/ComboBox.tsx

```diff
@@ -153,12 +153,11 @@ export default function ComboBox<ComboBoxOption extends ComboBoxItem>({
       setInputItems(filtered);
       // Always reset the first item to highlighted on filtering, to speed up selection.
       setHighlightedIndex(0);
+      // Hide any preview when the user starts typing.
+      setPreviewedIndex(-1);
```

Comment:
Oops, I just noticed one small bug here: this happens after an `if (!val)` check above that does an early return. This means if you do a filter, preview a block, and then empty out the search box in a single change (e.g. by pressing Cmd+Delete or Cmd+A then Delete), you'll end up with a preview of a different block.